### PR TITLE
Feat: Added Search functionality for lineage

### PR DESCRIPTION
**Pull request by ShaileshParmar11**
### Describe your changes :

I worked on the Adding Search functionality for the lineage also exit full screen functionality
#8521

#
### Type of change :

- [x] Improvement
- [x] New feature

#
### Frontend Preview (Screenshots) :

https://user-images.githubusercontent.com/71748675/200830333-d6012188-3c4c-4bba-ad8f-477d1b32302a.mov

https://user-images.githubusercontent.com/71748675/200832172-a575c58d-2f40-41b9-afa1-bfd76bfe4391.mov

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomControls.component.tsx [hunk 1]
```diff
@@ -136,90 +130,123 @@ const CustomControls: FC<ControlProps> = ({
...
+          allowClear
+          showSearch
+          filterOption={(input, option) =>
+            ((option?.label as string) ?? '')
+              .toLowerCase()
+              .includes(input.toLowerCase())
+          }
+          options={nodeOptions}
+          placeholder={t('label.search-lineage')}
+          style={{ width: 400, height: 32, marginLeft: isEditMode ? 83 : 0 }}
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomControls.component.tsx:148]: Can we use a class here?

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/EntityLineage.component.tsx [hunk 2]
```diff
@@ -1190,6 +1192,24 @@ const EntityLineageComponent: FunctionComponent<EntityLineageProp> = ({
...
+      const selectedNode = nodes.find((node) => node.id === value);
+
+      if (selectedNode) {
+        const { position } = selectedNode;
+        onNodeClick(selectedNode);
+        reactFlowInstance &&
+          reactFlowInstance.setCenter(position.x, position.y, {
+            duration: ZOOM_TRANSITION_DURATION,
+            zoom: MIN_ZOOM_VALUE,
+          });
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/EntityLineage.component.tsx:1206]: Add some comment here.

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomControls.component.tsx [hunk 3]
```diff
@@ -136,90 +130,123 @@ const CustomControls: FC<ControlProps> = ({
...
+              className={classNames(
+                'h-8 w-8 rounded-full p-x-xss tw-shadow-lg',
+                {
+                  'bg-primary': !isEditMode,
+                  'bg-primary-hover-lite': isEditMode,
+                }
+              )}
+              data-testid="edit-lineage"
+              disabled={!hasEditAccess}
+              title={hasEditAccess ? 'Edit Lineage' : NO_PERMISSION_FOR_ACTION}
```
[the remark this answers is not in the document]
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomControls.component.tsx:242]: done

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomControls.component.tsx [hunk 4]
```diff
@@ -136,90 +130,123 @@ const CustomControls: FC<ControlProps> = ({
...
+                max={MAX_ZOOM_VALUE}
+                min={MIN_ZOOM_VALUE}
+                step={ZOOM_SLIDER_STEP}
+                type="range"
+                value={zoom}
+                onChange={(e) => {
+                  const zoomValue = parseFloat(e.target.value);
+                  onZoomHandler(zoomValue);
+                  setZoom(zoomValue);
+                }}
```
[the remark this answers is not in the document]
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomControls.component.tsx:188]: done

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomControls.component.tsx (hunk 1 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomControls.component.tsx:148]: done

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomControls.component.tsx [hunk 5]
```diff
@@ -136,90 +130,123 @@ const CustomControls: FC<ControlProps> = ({
...
-      )}
-    </div>
+      <Col span={12}>
+        <Select
+          allowClear
+          showSearch
+          filterOption={(input, option) =>
+            ((option?.label as string) ?? '')
+              .toLowerCase()
+              .includes(input.toLowerCase())
```
[the remark this answers is not in the document]
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomControls.component.tsx:144]: done

**Comment by ShaileshParmar11** [comment]
Hi @Sachin-chaurasiya, All the comments are addressed!

**Review by Sachin-chaurasiya — APPROVED** [review]

**Pull request merged**